_modules/sesceph/purger.py

```diff
@@ -186,32 +185,30 @@ def purge(mdl, **kwargs):
     """
     service_shutdown_ceph()
     pur_ctrl = purger(mdl)
+    updater = mdl_updater.model_updater(mdl)
+    updater.hostname_refresh()
     try:
-        pur_ctrl.update_mon()
-    except mdl_updater.Error, e:
-        log.error(e)
+        updater.defaults_refresh()
     except utils.Error, e:
         log.error("exception self.updater.defaults_refresh()")
         log.error(e)
-    keyobj = keyring.keyring_facard(mdl)
-    for keytype in ["mds", "rgw", "osd", "mon", "admin"]:
+    if mdl.cluster_name == None:
+        log.error("Cluster name not found")
+    else:
         try:
-            keyobj.key_type = keytype
-
-        except keyring.error, E:
-            log.warning(E)
-            continue
-        if keyobj.present() is False:
-            log.info("Already removed '%s' keyring" % (keytype))
-            continue
-        log.info("Removing '%s' keyring" % (keytype))
-        keyobj.remove()
+            log.debug("Cluster name %s" % (mdl.cluster_name))
+            updater.load_confg(mdl.cluster_name)
+            updater.mon_members_refresh()
+        except mdl_updater.Error, e:
+            log.error(e)
+    pur_ctrl.auth_remove()
+    updater.symlinks_refresh()
+    updater.partitions_all_refresh()
```

Comment:
Do these three gracefully no-op if the cluster name is not found? Or should they be under the else?